src/pyproject_hooks/_impl.py

```diff
@@ -380,6 +381,15 @@ def build_sdist(
             },
         )
 
+    def _get_warnopts(self) -> Iterator[str]:
+        """
+        Reconstruct Python's warn options that are active for the current process,
+        so that it can be forwarded to a subprocess.
+        """
+        # `sys.warnoptions` is documented/mentioned in
+        # https://docs.python.org/3.13/library/warnings.html#describing-warning-filters
+        return chain.from_iterable(("-W", opt) for opt in sys.warnoptions)
```

Comment:
Python docs do mention that `sys.warnoptions` should not be modified but they do not prevent reading it.

In fact, docs for 3.9 even include examples using it for quick checks https://docs.python.org/3.9/library/warnings.html#overriding-the-default-filter:

```
import sys

if not sys.warnoptions:
    import warnings
    warnings.simplefilter("ignore")
```